tests/tensor_search/test_api.py

```diff
@@ -199,6 +200,75 @@ def test_search_timeout_short_timer_fails(self):
                         self.assertEqual(res.json()["code"], "vector_store_timeout")
                         self.assertEqual(res.json()["type"], "invalid_request")
 
+    def test_custom_search_limits(self):
+        custom_limit = 2000
+        custom_offset = 2000
+
+        with mock.patch.dict(os.environ, {
+            EnvVars.MARQO_MAX_SEARCH_LIMIT: str(custom_limit),
+            EnvVars.MARQO_MAX_SEARCH_OFFSET: str(custom_offset)
+        }):
+            importlib.reload(sys.modules['marqo.tensor_search.api'])
+
+            # Explicitly call generate_config to trigger initialization
+            new_config = api.generate_config()
+
+            # Initialize a new IndexManagement object with the new config
+            index_management = IndexManagement(new_config.vespa_client)
+
+            # Mock the open function to capture the written content
+            mock_open_result = mock.mock_open()
+            with mock.patch('builtins.open', mock_open_result):
+                index_management._add_default_query_profile('dummy_path')
+
+            # Get the written content
+            written_content = mock_open_result().write.call_args[0][0]
+
+            # Check if the custom limits were correctly set in the query profile
+            self.assertIn(f'<field name="maxHits">{custom_limit}</field>', written_content)
+            self.assertIn(f'<field name="maxOffset">{custom_offset}</field>', written_content)
+
+            # Test the actual api
+            self.client = TestClient(api.app)
+
+            with self.subTest(search_method="TENSOR"):
+                for index in [self.unstructured_index, self.structured_index]:
+                    with self.subTest(index=index.name):
+                        res = self.client.post("/indexes/" + index.name + "/search?device=cpu", json={
+                            "q": "irrelevant",
+                            "searchMethod": "TENSOR",
+                            "limit": custom_limit,
+                            "offset": custom_offset,
+                        })
+                        self.assertEqual(res.status_code, 200)
+
+                        res = self.client.post("/indexes/" + index.name + "/search?device=cpu", json={
+                            "q": "irrelevant",
+                            "searchMethod": "TENSOR",
+                            "limit": custom_limit + 1,
+                            "offset": custom_offset + 1,
+                        })
+                        self.assertEqual(res.status_code, 400)
+
+            with self.subTest(search_method="HYBRID"):
```

Comment:
you can execute this in a loop `for search_method in ["TENSOR", "HYBRID"]:` so you don't repeat the code

Reply:
Good point! Will refactor in the morning

Reply:
Does Hybrid support Unstructured now though?

Reply:
test removed